File: 1-two-sum/1-two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seenValues = new HashMap<Integer, Integer>();
        int i = 0;
        for (int num : nums) {
            if (seenValues.get(target - num) != null) {
                return new int[] { i, seenValues.get(target - num) };
            }
            seenValues.put(num, i);
            i++;
        }
        return new int[] { -1, 1 };
    }
}

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
    
        dict = {}

        i = 0
        for num in nums:
            diff = target - num
            if diff in dict:
                return [i, dict[diff]]
            dict[num] = i
            i += 1